pkg/util/sys/device.go

```diff
@@ -43,6 +44,14 @@ type Partition struct {
 	Filesystem string
 }
 
+type CephVolumeDeviceData struct {
```

Comment:
If you add the json properties to the struct like the following, I would expect the deserialization will be automated. 
```
type CephVolumeDeviceData struct {
	DevicePath string `json:"devicePath"`
...
}
```

Reply:
The problem is that the info provided by `ceph-volume inventory --format json` is much more deeper that the info we really need.( the dashboard and external clients use the information provided by the command `ceph-volume inventory` which provides only the four fields that we have in our CephVolumeDeviceData struct.

As i said, we only need a few fields from all the info provided ( the fields that we have in our CephVolumeDeviceData struct). Besides this fact, the fields that we need are in two different json levels, `available` is in the first level, but the rest of the fields are inside the `sys_api` level.
Another issue is that most of the info provided by ceph-volume is already present in the devices configmap, so adding "repeated" information seems to be not a good idea.

Taking into account the previous facts. In my opinion, this implementation has several advantages:
- Include in the configmap only the information is required.
- Do not overload with repeated information the configmap
- Not heavily linked to the json output provided by ceph-volume inventory
- Easy to adapt/include/remove fields as needed
- provide directly the ceph-volume fields required as json fields, not as embedded in a json string that must be processed to get the info.
 

**Details about output provide by ceph-volume:**
Example of `ceph-volume inventory` (this is what clients expects, for example the dashboard, or the ceph orchestrator)
```
# ceph-volume inventory

Device Path               Size         rotates available Model name
/dev/sdb                  25.00 GB     True    True      VBOX HARDDISK
/dev/sda                  32.00 GB     True    False     VBOX HARDDISK
```

**However, the output provided in json format is wider:**
```
# ceph-volume inventory --format json
[{"available": true, "rejected_reasons": [], "sys_api": {"scheduler_mode": "noop", "rotational": "1", "vendor": "ATA", "human_readable_size": "25.00 GB", "sectors": 0, "sas_device_handle": "", "partitions": {}, "rev": "1.0", "sas_address": "", "locked": 0, "sectorsize": "512", "removable": "0", "path": "/dev/sdb", "support_discard": "", "model": "VBOX HARDDISK", "ro": "0", "nr_requests": "128", "size": 26843545600.0}, "lvs": [], "path": "/dev/sdb"}, {"available": false, "rejected_reasons": ["locked"], "sys_api": {"scheduler_mode": "noop", "rotational": "1", "vendor": "ATA", "human_readable_size": "32.00 GB", "sectors": 0, "sas_device_handle": "", "partitions": {"sda2": {"start": "2099200", "holders": ["dm-0", "dm-1"], "sectorsize": 512, "sectors": "65009664", "size": "31.00 GB"}, "sda1": {"start": "2048", "holders": [], "sectorsize": 512, "sectors": "2097152", "size": "1024.00 MB"}}, "rev": "1.0", "sas_address": "", "locked": 1, "sectorsize": "512", "removable": "0", "path": "/dev/sda", "support_discard": "", "model": "VBOX HARDDISK", "ro": "0", "nr_requests": "128", "size": 34359738368.0}, "lvs": [{"comment": "not used by ceph", "name": "swap"}, {"comment": "not used by ceph", "name": "root"}], "path": "/dev/sda"}]

```

Reply:
Ok, let's separate this into two different issues:
1. Deserialization from ceph-volume: We shouldn't need any code to deserialize json. We should declare the needed struct and let the json.Unmarshal() method do all the hard work. The nice thing is that if you only define the properties you care about, the json deserialization will ignore the rest. Then you can just copy the needed values from the deserialization struct into your `CephVolumeDeviceData` struct.
1. Serialize into the configmap: The CephVolumeDeviceData contains only the properties needed. I'd suggest you add the json attributes to the struct so we ensure what json will be generated.

Reply:
Thank you!!. Your suggestion really simplifies heavily this part. Learned some things about json management in golang. :-)